### PR TITLE
Use state and mirage  

### PR DESCRIPTION
**Pull request by DM0000**
added mirage
useState to track UserPosts
eslint and propTypes
cleanup

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -12,9 +12,11 @@
   "dependencies": {
     "@faker-js/faker": "^9.0.0",
     "bootstrap": "^5.3.3",
-    "popper.js": "^1.16.1",
     "husky": "^9.1.6",
+    "popper.js": "^1.16.1",
```
Comment (rdkerr) [inline package.json]: Using this?

File: src/Components/ArticleStub.jsx [hunk 2]
```diff
@@ -1,60 +1,46 @@
...
+    <Accordion.Item>
+      <Accordion.Header>
+        <img src={bread} />
+        <h2>
+          {" "}
+          <Person name={name} country={country} />
+        </h2>
+      </Accordion.Header>
+      <Accordion.Body>
+        <strong>This is the second item's accordion body.</strong> It is hidden
```
Comment (rdkerr) [inline src/Components/ArticleStub.jsx]: Could replace this with some faker paragraphs

File: src/Components/Content.jsx [hunk 3]
```diff
@@ -1,20 +1,20 @@
-import PostList from "./PostList"
+import PostList from "./PostList";
+import PropTypes from "prop-types";
 
+function Content({ initialPostCount }) {
+  return (
+    <div className='content'>
+      <div className='heading'>
```
Comment (rdkerr) [inline src/Components/Content.jsx:8]: Do we need this Div? Coudl we just apply "heading" to the following h1?

File: src/Components/PostList.jsx [hunk 4]
```diff
@@ -1,15 +1,39 @@
-import FakeArticleStub from "./ArticleStub.jsx";
+import DropDown from "./ArticleStub.jsx";
+import { useState, useEffect } from "react";
+import Accordion from "react-bootstrap/Accordion";
+import PropTypes from "prop-types";
 
 //TODo: update to take stubProp
```
Comment (rdkerr) [inline src/Components/PostList.jsx]: Safe to remove now that you added the prop?

File: src/main.jsx [hunk 5]
```diff
@@ -1,12 +1,10 @@
-import { StrictMode } from "react";
+// import { StrictMode } from "react";
```
Comment (rdkerr) [inline src/main.jsx]: We probably shouldn't disable this.  You could implement this wrapper if you don't want to deal with the double calls on mount: https://taig.medium.com/prevent-react-from-triggering-useeffect-twice-307a475714d7

Comment (rdkerr) [reply src/main.jsx]: Actually thats gross, don't implement that

Comment (rdkerr) [reply src/main.jsx]: We will eventually swap to rtk-query which will eliminate this as a problem

**Comment by DM0000** [comment]
Think I can add it back in now that server is out of app it should be ok.  Sent from my iPhoneOn Nov 19, 2024, at 2:46 PM, Roger Kerr ***@***.***> wrote:﻿
@rdkerr commented on this pull request.



In src/main.jsx:
> @@ -1,12 +1,10 @@
-import { StrictMode } from "react";
+// import { StrictMode } from "react";

We will eventually swap to rtk-query which will eliminate this as a problem

File: package.json (hunk 1 above)
Comment (DM0000) [reply package.json]: no its for positioning, ill remove for now

File: src/Components/ArticleStub.jsx (hunk 2 above)
Comment (DM0000) [reply src/Components/ArticleStub.jsx]: ok added

File: src/Components/Content.jsx (hunk 3 above)
Comment (DM0000) [reply src/Components/Content.jsx:8]: not atm, currently breaks it.   will add todo to fix when I do  CSS/react pass

File: src/Components/PostList.jsx (hunk 4 above)
Comment (DM0000) [reply src/Components/PostList.jsx]: done

File: src/main.jsx (hunk 5 above)
Comment (DM0000) [reply src/main.jsx]: reapplied StrictMode

File: src/server/fakeServer.jsx [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+export default function startServer() {
+  createServer({
+    routes() {
+      this.namespace = "api";
+      this.timing = 2000;
+
+      this.get("/users/:amount", (_, request) => {
+        let amount = request.params.amount;
+
+        let x = generatePosts(amount);
```
Comment (rdkerr) [inline src/server/fakeServer.jsx]: #nit 
let posts = generatePosts

File: src/server/fakeServer.jsx [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+
+        let y = JSON.stringify(x);
```
Comment (rdkerr) [inline src/server/fakeServer.jsx]: You're not using the variable y for anything.  Just return Json.string(x)

File: src/server/fakeServer.jsx (hunk 6 above)
Comment (DM0000) [reply src/server/fakeServer.jsx]: ok fixed and chanded json post to jsonPost from y

File: src/server/fakeServer.jsx (hunk 7 above)
Comment (DM0000) [reply src/server/fakeServer.jsx]: haha just read this , fixed

**Review by rdkerr — APPROVED** [review]

**Pull request merged**